diff --git a/src/rerun/components/video_codec.cpp b/src/rerun/components/video_codec.cpp
index 85cb0b6..9f1b5fe 100644
--- a/src/rerun/components/video_codec.cpp
+++ b/src/rerun/components/video_codec.cpp
@@ -48,7 +48,7 @@ namespace rerun {
         ARROW_RETURN_NOT_OK(builder->Reserve(static_cast<int64_t>(num_elements)));
         for (size_t elem_idx = 0; elem_idx < num_elements; elem_idx += 1) {
             const auto variant = elements[elem_idx];
-            ARROW_RETURN_NOT_OK(builder->Append(static_cast<uint8_t>(variant)));
+            ARROW_RETURN_NOT_OK(builder->Append(static_cast<uint32_t>(variant)));
         }
 
         return Error::ok();
